modules/betweenBidAdapter.js

```diff
@@ -13,7 +14,22 @@ export const spec = {
    * @return boolean True  if this is a valid bid, and false otherwise.
    */
   isBidRequestValid: function(bid) {
-    return !!(bid.params.w && bid.params.h && bid.params.s);
+    if (bid && bid.params) {
+      if (typeof bid.params !== 'object' || Array.isArray(bid.params)) {
+        utils.logError(BIDDER_CODE + ': the type of bid.params is not object.');
+        return false;
+      } else if (!(bid.params.w && bid.params.h && bid.params.s) ||
+        !utils.isNumber(bid.params.w) || !utils.isNumber(bid.params.h) ||
+        !utils.isNumber(bid.params.s)) {
+        utils.logError(BIDDER_CODE + ': the value of adUnits params is wrong or absent.');
+        return false;
+      } else {
+        return true;
+      }
+    } else {
+      utils.logError(BIDDER_CODE + ': empty bid object.');
+      return false;
+    }
```

Comment:
This looks fine,

But perhaps it could be simplified a little.

You can utilize the `utils.deepAccess` method and get away with some of this. Here is a quick example:

```javascript
isBidRequestValid: function(bid) {
   let params = utils.deepAccess(bid, 'params') || {};
   if (utils.isNumber(params.w) && utils.isNumber(params.h) && utils.isNumber(params.s)) {
      return true;
   } else {
      utils.logError(`The value of ${BIDDER_CODE}'s params are incorrect. 'w', 'h', and 's' must all be numbers`);
      return false;
   }
}
```

Although we do like it if an adapter is flexible when it comes to hard requiring params to be a number rather than checking if the passed param as a string is actually just a number represented as a string. Sometimes publishers / users will put all bidder params as `Strings` even when the value of your param is technically a number:

```javascript
params = {
   w: '728',
   h: '90',
   s: '3245235'
}
```

Technically would not work, when the intention is there.

So you _could_ add some logic in your adapter to convert to numbers if passed as string, but no worries, not required, just a thought.

But regardless, I think simplifying this into one if / else is more clean and readable.